python-package/xgboost/sklearn.py

```diff
@@ -598,28 +604,8 @@ class XGBClassifier(XGBModel, XGBClassifierBase):
     __doc__ = "Implementation of the scikit-learn API for XGBoost classification.\n\n" \
         + '\n'.join(XGBModel.__doc__.split('\n')[2:])
 
-    def __init__(self, max_depth=3, learning_rate=0.1,
-                 n_estimators=100, verbosity=1,
-                 objective="binary:logistic", booster='gbtree',
-                 tree_method='auto', n_jobs=1, gpu_id=-1, gamma=0,
-                 min_child_weight=1, max_delta_step=0, subsample=1,
-                 colsample_bytree=1, colsample_bylevel=1, colsample_bynode=1,
-                 reg_alpha=0, reg_lambda=1, scale_pos_weight=1, base_score=0.5,
-                 random_state=0, missing=None, **kwargs):
-        super(XGBClassifier, self).__init__(
-            max_depth=max_depth, learning_rate=learning_rate,
-            n_estimators=n_estimators, verbosity=verbosity,
-            objective=objective, booster=booster, tree_method=tree_method,
-            n_jobs=n_jobs, gpu_id=gpu_id, gamma=gamma,
-            min_child_weight=min_child_weight,
-            max_delta_step=max_delta_step, subsample=subsample,
-            colsample_bytree=colsample_bytree,
-            colsample_bylevel=colsample_bylevel,
-            colsample_bynode=colsample_bynode,
-            reg_alpha=reg_alpha, reg_lambda=reg_lambda,
-            scale_pos_weight=scale_pos_weight,
-            base_score=base_score, random_state=random_state, missing=missing,
-            **kwargs)
+    def __init__(self, objective="binary:logistic", **kwargs):
```

Comment:
I looked into the source code of inspection, not sure if this might break sklearn grid search.  Will help adding some tests here.

Reply:
`test_kwargs_grid_search` for `XGBClassifier` and `test_parameter_tuning` for `XGBRegressor` uses `GridSearchCV`. Please let me know if you have more specific case in my mind. 
Btw. just an idea for separate task if not already done: compare XGBoost test-cases with LightGBM for Sklearn - a lot of overlap, but maybe there are interesting cases that  are missing.

Reply:
Definitely.  I'm seeing more and more users preferring sklearn interface over more primitive train and predict.  Will try to make it my default interface for development in the future.